repos/system_upgrade/common/libraries/tests/test_rhsm.py

```diff
@@ -119,3 +209,169 @@ def test_inhibit_on_duplicate_repos_no_dups(monkeypatch):
+    +----------------------------------------------------------+
+    '''
+
+    for enabled_repository in RHSM_ENABLED_REPOS:
+        rhsm_output_fragment = 'Repo ID: {0}\n'.format(enabled_repository.id)
+        rhsm_output_fragment += 'Repo Name: {0}\n'.format(enabled_repository.name)
+        rhsm_output_fragment += 'Repo URL: {0}\n'.format(enabled_repository.url)
+        rhsm_output_fragment += 'Enabled: {0}\n'.format(enabled_repository.enabled)
+        rhsm_output_fragment += '\n'
+        rhsm_list_enabled_output += rhsm_output_fragment
+
+    context_mocked.add_mocked_command_call_with_stdout(RHSM_LIST_ENABLED_REPOS, rhsm_list_enabled_output)
+
+    enabled_repo_ids = rhsm.get_enabled_repo_ids(context_mocked)
+
+    fail_description = 'Failed to detected enabled repositories on the system.'
+    assert len(enabled_repo_ids) == 3, fail_description
+
+    fail_description = 'Failed to retrieve repository ID provided in the RHSM output.'
+    for enabled_repository in RHSM_ENABLED_REPOS:
+        assert enabled_repository.id in enabled_repo_ids, fail_description
+
+
+def test_get_existing_product_certificates(monkeypatch, actor_mocked, context_mocked):
+    """Verifies that the library is able to correctly retrieve existing product certificates."""
+
+    def mocked_isdir(path):
+        # Only return True for these two specific paths specific to this test
+        if path == '/etc/pki/product-default':
+            return True
+        if path == '/etc/pki/product':
+            return False
+        err_message = 'RHSM library should not gather info about additional dirs (attempted to isdir: {0}).'
+        raise ValueError(err_message.format(path))
+
+    # Describe the fictional certificate directory layout
+    product_default_directory_contents = {
+        'files': ['cert2', 'cert3'],
+        'dirs': ['cert4']
+    }
+
+    def mocked_listdir(path):
+        if path == '/etc/pki/product':
+            # We pretend that /etc/pki/product is a file, not a dir (some weird system state)
+            raise OSError(os.errno.ENOTDIR, 'Attempting to listdir a file: {0}'.format(path))
+        if path == '/etc/pki/product-default':
+            return product_default_directory_contents['files'] + product_default_directory_contents['dirs']
+
+        err_message = 'RHSM library should not listdir additional dirs (attempted to listdir: {0}).'
+        raise ValueError(err_message.format(path))
+
+    def mocked_isfile(path):
+        if path == '/etc/pki/product':
+            return True
+        if path == '/etc/pki/product-default':
+            return False
+
+        # Handle imaginary files inside the '/etc/pki/product-default'
+        dirname = os.path.dirname(path)
+        if dirname == '/etc/pki/product-default':
+            filename = os.path.basename(path)
+            if filename in product_default_directory_contents['files']:
+                return True
+            return False
```

Comment:
Regarding the amount of the code, wouldn't be a better to have just a list of paths which returns true? E.g.
```
ISFILE_PATHS = [
    '/etc/pki/product',
    '/etc/pki/product-default/cert2',
    '/etc/pki/product-default/cert3',
]
if path in ISFILE_PATHS:
    return True
if path == '/etc/pki/product-default'
   return False
```
I guess we will not extend `product_default_directory_contents` significantly in future.